packages/markdown-docx/src/OOXMLTransformer/index.js

```diff
@@ -58,171 +87,178 @@ class OoxmlTransformer {
-     * @returns {string} the value of the variable
+     * @param {object} nodeInformation Contains properties and value of a node
+     * @return {object} CiceroMark Node
      */
-    getValue(elements) {
-        const variableContents = elements[1];
-        for (const property of variableContents.elements[0].elements) {
-            if (property.name === 'w:t') {
-                return property.elements[0].text;
-            }
+    construtCiceroMarkNodeJSON(nodeInformation) {
+        let obj = {};
+        if (nodeInformation.nodeType === 'softbreak') {
+            obj = {
+                $class: `${NS_PREFIX_CommonMarkModel}Softbreak`,
+            };
+        } else if (nodeInformation.nodeType === 'variable') {
+            obj = {
+                $class: `${NS_PREFIX_CiceroMarkModel}Variable`,
+                value: nodeInformation.value || 'Not provided',
+                elementType: nodeInformation.elementType || 'Not a template variable',
+                name: nodeInformation.name || 'No name available',
+            };
+        } else {
+            obj = {
+                $class: `${NS_PREFIX_CommonMarkModel}Text`,
+                text: nodeInformation.value,
+            };
         }
-        return null;
+        for (let i = nodeInformation.properties.length - 1; i >= 0; i--) {
+            obj = {
+                $class: nodeInformation.properties[i],
+                nodes: [obj],
+            };
+        }
+        return obj;
     }
 
     /**
-     * Checks if the element is a line break or not
+     * Creates a pargraph, heading or block CiceroMark Node
      *
-     * @param {object} element the element to be checked
-     * @returns {boolean} whether the element is a line break
+     * @param {object} rootNode Pargraph, heading or any other node
      */
-    isLineBreak(element) {
-        return element.name === 'w:p' && element.elements === undefined;
-    }
+    constructNodes(rootNode) {
+        if (this.JSONXML.length > 0) {
+            let constructedNode;
+            constructedNode = this.construtCiceroMarkNodeJSON(this.JSONXML[0]);
+            rootNode.nodes = [...rootNode.nodes, constructedNode];
 
-    /**
-     * Returns if the element is a heading or not and if it is, an attribute
-     * level is also passed to determine the style of the heading.
-     *
-     * @param {object} headingElement the element to be checked
-     * @returns {object} object of `isHeading` and its level
-     */
-    getHeading(headingElement) {
-        if (headingElement.attributes !== undefined) {
-            const headingLevel = headingElement.attributes['w:val'];
-            if (headingElement.name === 'w:pStyle' && headingLevel.includes('Heading')) {
-                return {
-                    isHeading: true,
-                    level: headingLevel[headingLevel.length - 1],
+            let rootNodesLength = 1;
+            for (let i = 1; i < this.JSONXML.length; i++) {
```

Comment:
What exactly `i` is? Could you rename it so that it makes more sense?

Reply:
`i` is the index here for ith element. I will rename it to `index`